src/scores/emerging/risk_matrix.py

```diff
@@ -0,0 +1,497 @@
+        and severity_dim.
+
+    Raises:
+        ValueError: if ``matrix_weights`` isn't two dimensional.
+        ValueError: if number of rows of ``matrix_weights`` doesn't equal length of ``prob_threshold_coords``.
+        ValueError: if number of columns of ``matrix_weights`` doesn't equal length of ``severity_coords``.
+        ValueError: if ``prob_threshold_coords`` aren't strictly between 0 and 1.
+
+    References:
+        - Taggart, R. J., & Wilke, D. J. (2024). Warnings based on risk matrices: a coherent framework with consistent evaluation.
+            In preparation.
+
+    Examples:
+        Returns risk matrix decision weights, where weights increase with increasing
+        severity category and decrease with increasing probability threshold.
+
+        >>> import numpy as np
+        >>> from scores.emerging import matrix_weights_to_array
+        >>> matrix_weights = np.array([
+        >>>     [1, 2, 3],
+        >>>     [2, 4, 6],
+        >>>     [3, 6, 9],
+        >>> ])
+        >>> matrix_weights_to_array(
+        >>>     matrix_weights, "severity", ["MOD+", "SEV+", "EXT"], "prob_threshold", [0.1, 0.3, 0.5]
+        >>> )
+    """
+    matrix_dims = matrix_weights.shape
+
+    if len(matrix_dims) != 2:
+        raise ValueError("`matrix_weights` must be two dimensional")
+    if matrix_dims[0] != len(prob_threshold_coords):  # type: ignore
+        raise ValueError("number of `prob_threshold_coords` must equal number of rows of `matrix_weights`")
+    if matrix_dims[1] != len(severity_coords):  # type: ignore
+        raise ValueError("number of `severity_coords` must equal number of columns of `matrix_weights`")
+
+    if (np.max(prob_threshold_coords) >= 1) or (np.min(prob_threshold_coords) <= 0):
+        raise ValueError("`prob_threshold_coords` must strictly between 0 and 1")
+
+    prob_threshold_coords = np.flip(np.sort(np.array(prob_threshold_coords)))
+
+    weight_matrix = xr.DataArray(
+        data=matrix_weights,
+        dims=[prob_threshold_dim, severity_dim],
+        coords={
+            prob_threshold_dim: prob_threshold_coords,
+            severity_dim: severity_coords,
+        },
+    )
+
+    return weight_matrix
+
+
+def weights_from_warning_scaling(
+    scaling_matrix: np.ndarray,
+    assessment_weights: Iterable,
+    severity_dim: str,
+    severity_coords: Iterable,
+    prob_threshold_dim: str,
+    prob_threshold_coords: Iterable,
+) -> xr.DataArray:
+    """
+    Caution:
+        This function is part of an emerging score that is under development and has not yet
+        been peer-reviewed. It may change at any time.
+
+    Given a warning scaling matrix, assessment weights and other inputs,
+    returns the decision weights for the :py:func:`scores.emerging.risk_matrix_score` as an xarray data array.
+
+    Comprehensive checks are made on ``scaling_matrix`` to ensure it satisfies the properties
+    of warning scaling in Table 1 of Taggart & Wilke (2004).
```

Comment:
```suggestion
    of warning scaling in Table 1 of Taggart & Wilke (2024).
```
Is this meant to be Taggart & Wilke (2024)?